File: src/main/java/org/jcodec/api/JCodecException.java
package org.jcodec.api;

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 * 
 * An exception thrown by JCodec API functions in case something goes wrong
 * 
 * @author The JCodec Project
 * 
 */
public class JCodecException extends Exception {

    public JCodecException(String message) {
        super(message);
    }

    public JCodecException(String message, Exception cause) {
        super(message, cause);
    }
}

File: src/main/java/org/jcodec/api/JCodecException.kt
package org.jcodec.api

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 *
 * @author The JCodec project
 */
open class JCodecException : Exception {
    constructor(arg0: String?) : super(arg0) {}
    constructor(e: Throwable?) : super(e) {}
}